Task/add silent entities check time limit (#827)
telefonicaid/perseo-fe: docs/API/plain_rules.md

    values are truncated to them (a warning log message is generated if such truncation occurs)
-   **attribute**: _mandatory_, attribute for watch
-   **reportInterval**: _mandatory_, time in seconds to see an entity as silent (in other words, how much time has to
    pass since last entity update to consider that entity is in "no signal" situation so the rule is triggered). In
    order to not check too old entities there is a _maxTimeDetection_ (perseo checks for the attribute first in the
    entity configuration, maxTimeDetectionAttr, then in the rule configuration, and lastly, if not found, in the global
    configuration) to limit this search
-   **reportIntervalAttr**: _optional_, entity attribute where find value for reportInterval.
-   **maxTimeDetection**: _optional_, max time in seconds to see an entity as silent (in other words, how much time has
    to pass since last entity update to not consider that entity is in "no signal" situation due to entity is death (no
    modified during a long time)

> -   **maxTimeDetectionAttr**: _optional_, entity attribute where find value for maxTimeDetection.

Reviewer: Maybe this new fields (3) in this kind of rules should be mentioned in CNR?

Reviewer: In addition, being all 3 optional attributes, it should be specified what happen when they are not provided. In particular:

* In which attribute is reportInternval when reportInterval is not provided?
* Which value takes the max detection time when maxTimeDetection is not provided? (I guess that the one in PERSEO_NON_SIGNAL_MAX_TIME_DETECTION, but it is not explicitly said here)
* Which attribute is used when maxTimeDetectionAttr is not used? (In fact, I'm not sure that means this attribute... as maxTimeDetection is a field in the rule...)

Author: Explained here: https://github.com/telefonicaid/perseo-fe/pull/827/commits/034b91072ed079ed1ad0c7a73770bb46471757cd

    values are truncated to them (a warning log message is generated if such truncation occurs)
-   **attribute**: _mandatory_, attribute for watch
-   **reportInterval**: _mandatory_, time in seconds to see an entity as silent (in other words, how much time has to
    pass since last entity update to consider that entity is in "no signal" situation so the rule is triggered). In
    order to not check too old entities there is a _maxTimeDetection_ (perseo checks for the attribute first in the
    entity configuration, maxTimeDetectionAttr, then in the rule configuration, and lastly, if not found, in the global
    configuration) to limit this search

> -   **reportIntervalAttr**: _optional_, entity attribute where find value for reportInterval. If no _reportIntervalAttr_
> -   is found then _reportInterval_ from rule is having in to account. 
> -   **maxTimeDetection**: _optional_, max time in seconds to see an entity as silent (in other words, how much time has
>     to pass since last entity update to not consider that entity is in "no signal" situation due to entity is death (no
>     modified during a long time). If no value is provided then Perseo global config (nonSignalMaxTimeDetection or 
>     docker env var PERSEO_NON_SIGNAL_MAX_TIME_DETECTION) is having into account. By default nonSignalMaxTimeDetection is
>     set to 30 days.
> -   **maxTimeDetectionAttr**: _optional_, entity attribute where find value for maxTimeDetection. If no _maxTimeDetectionAttr_
> -   is found then _maxTimeDetection_ from rule is having in to account.

Reviewer: ```suggestion
-   **reportIntervalAttr**: _optional_, entity attribute where find value for reportInterval. If no _reportIntervalAttr_
-   is found then _reportInterval_ from rule is used. 
-   **maxTimeDetection**: _optional_, max time in seconds to see an entity as silent (in other words, how much time has
    to pass since last entity update to not consider that entity is in "no signal" situation due to entity is death, i.e. not
    modified during a long time). If no value is provided then Perseo global config (nonSignalMaxTimeDetection or 
    docker env var PERSEO_NON_SIGNAL_MAX_TIME_DETECTION) is used. By default nonSignalMaxTimeDetection/PERSEO_NON_SIGNAL_MAX_TIME_DETECTION is
    set to 30 days.
-   **maxTimeDetectionAttr**: _optional_, entity attribute where find value for maxTimeDetection. If no _maxTimeDetectionAttr_
-   is found then _maxTimeDetection_ from rule is used.
```

-   **ruleName**: name of the rule
-   **reportInterval**: time to see an entity as silent (in other words, how much time has to pass since last entity
    update to consider that entity is in "no signal" situation so the rule is triggered)
-   **reportIntervalAttr**: entity attribute where find value for reportInterval.
-   **maxTimeDetection**: max time in seconds to not see an entity as silent (in other words, how much time has to pass
    since last entity update to not consider that entity is in "no signal" situation due to entity is death (no modified
    during a long time)

> -   **maxTimeDetectionAttr**: entity attribute where find value for maxTimeDetection.

Author: these condition fields are also added to action part.

> -   **maxTimeDetectionAttr**: entity attribute where find value for maxTimeDetection.

Reviewer: They take the same values a in the condition of the rule?

> -   **maxTimeDetectionAttr**: entity attribute where find value for maxTimeDetection.

Author: They take the values from condtion of the rule

-   **ruleName**: name of the rule

> -   **reportInterval**: time to see an entity as silent (in other words, how much time has to pass since last entity
>     update to consider that entity is in "no signal" situation so the rule is triggered)
> -   **reportIntervalAttr**: entity attribute where find value for reportInterval.
> -   **maxTimeDetection**: max time in seconds to not see an entity as silent (in other words, how much time has to pass
>     since last entity update to not consider that entity is in "no signal" situation due to entity is death (no modified
>     during a long time)
> -   **maxTimeDetectionAttr**: entity attribute where find value for maxTimeDetection.

Reviewer: ```suggestion
-   **reportInterval**: time to see an entity as silent (in other words, how much time has to pass since last entity
    update to consider that entity is in "no signal" situation so the rule is triggered) (same value as field of the same field in rule condition).
-   **reportIntervalAttr**: entity attribute where find value for reportInterval (same value as field of the same field in rule condition).
-   **maxTimeDetection**: max time in seconds to not see an entity as silent (in other words, how much time has to pass
    since last entity update to not consider that entity is in "no signal" situation due to entity is death (no modified
    during a long time) (same value as field of the same field in rule condition)
-   **maxTimeDetectionAttr**: entity attribute where find value for maxTimeDetection (same value as field of the same field in rule condition).
```

I understand that if reportIntervalAttr/maxTimeDetection/maxTimeDetectionAttr is not specified in the condition (they are optional fields) they are not available in  non signal template action. Is my understanding correct?
